src/main/java/org/folio/domain/strategy/ItemStatusBasedTenantPickingStrategy.java

```diff
@@ -38,22 +33,31 @@ public class ItemStatusBasedTenantPickingStrategy implements TenantPickingStrate
   private final SearchClient searchClient;
 
   @Override
-  public Optional<String> pickTenant(String instanceId) {
+  public List<String> pickTenants(String instanceId) {
     log.info("pickTenant:: picking tenant for a TLR for instance {}", instanceId);
 
     var itemStatusOccurrencesByTenant = getItemStatusOccurrencesByTenant(instanceId);
     log.info("pickTenant:: item status occurrences by tenant: {}", itemStatusOccurrencesByTenant);
 
-    Optional<String> tenantId = Optional.<String>empty()
-      .or(() -> pickTenant(itemStatusOccurrencesByTenant, EnumSet.of(AVAILABLE)))
-      .or(() -> pickTenant(itemStatusOccurrencesByTenant, EnumSet.of(CHECKED_OUT, IN_TRANSIT)))
-      .or(() -> pickTenant(itemStatusOccurrencesByTenant, alwaysTrue())); // any status
+    List<String> sortedTenantIds = itemStatusOccurrencesByTenant.entrySet()
+      .stream()
+      .sorted(
+        comparingLong((Entry<String, Map<String, Long>> entry) -> countItems(
+            entry, Set.of("Available")::contains)).reversed()
+          .thenComparing(comparingLong((Entry<String, Map<String, Long>> entry) -> countItems(
+            entry, Set.of("Checked out", "In transit")::contains)).reversed())
+          .thenComparing(comparingLong((Entry<String, Map<String, Long>> entry) -> countItems(
+            entry, s -> true)).reversed()))
+      .map(Entry::getKey)
+      .toList();
```

Comment:
How about using enum for item statuses and making this comparator a bit prettier?
```
    List<String> sortedTenantIds = itemStatusOccurrencesByTenant.entrySet()
      .stream()
      .sorted(compareByItemCount(EnumSet.of(AVAILABLE))
        .thenComparing(compareByItemCount(EnumSet.of(CHECKED_OUT, IN_TRANSIT)))
        .thenComparing(compareByItemCount(alwaysTrue())))
      .map(Entry::getKey)
      .toList();

  private static Comparator<Entry<String, Map<String, Long>>> compareByItemCount(
    EnumSet<ItemStatusEnum> desiredStatuses) {

    Set<String> statusStrings = desiredStatuses.stream()
      .map(ItemStatusEnum::getValue)
      .collect(toSet());

    return compareByItemCount(statusStrings::contains);
  }

  private static Comparator<Entry<String, Map<String, Long>>> compareByItemCount(
    Predicate<String> statusFilter) {

    return comparingLong((Entry<String, Map<String, Long>> entry) -> entry.getValue()
      .entrySet()
      .stream()
      .filter(e -> statusFilter.test(e.getKey()))
      .map(Entry::getValue)
      .reduce(0L, Long::sum))
      .reversed();
  }
```